### ISSUE TITLE
关于bound的计算

### ISSUE THREAD

**Issue by HHCorp** [zhangkai0425/mipnerf-360#2]
感谢你的项目，相对于其他开源代码更加简洁，对于入门学习非常有帮助。

不过对于bound的计算，我的看法是：论文中和你的代码实现其实是一样的，惩罚高出proposal网络产生的直方图的部分，最终希望proposal网络的直方图“包裹”nerf的直方图。只不过论文中是在nerf的采样区间上进行，而你的代码是在proposal的等间隔采样区间上进行，最终的效果应该仍是相近的。不知道我的理解对不对，如果不对希望你能指正。

另外，bound计算部分的代码实现似乎不对：
`torch.sum(prop_weights[..., ~((t0>upper)|(t1<lower))], dim=-1)`输出的并不是期望的“batch中各个ray采样区间上prop_weight的和”，而是“整个batch所有ray采样区间上prop_weight的和”

**Comment by zhangkai0425** [comment]
谢谢你的仔细检查！非常抱歉，由于前几天一直在忙本科毕设，没有来得及回复！我觉得你说的应该是对的，当时bound是实现上确实有一定的区别，但是最终效果我也没有检验是否一致，如果你做了实验可以讲一下结果！后面的实现问题应该也是有问题，我之后会去改正的。(但是现在不做CV了QAQ，不知道啥时候改)

**Comment by HHCorp** [comment]
> 谢谢你的仔细检查！非常抱歉，由于前几天一直在忙本科毕设，没有来得及回复！我觉得你说的应该是对的，当时bound是实现上确实有一定的区别，但是最终效果我也没有检验是否一致，如果你做了实验可以讲一下结果！后面的实现问题应该也是有问题，我之后会去改正的。(但是现在不做CV了QAQ，不知道啥时候改)

哈哈，我主业也不是3D的，最近是刚开始入门nerf，只是在尝试用mip-360和instant-ngp的思路优化下采样点给普通nerf提提速，所以还并没做到mip-360的实验，之后开始复现360了再对比实验下。bound的实现我后来参考了这个https://github.com/Enigmatisms/NeRF/blob/master/nerf/addtional.py
